frontend/src/Card.tsx

```diff
@@ -1,16 +1,19 @@
-import { ReactComponent as Diamond } from './assets/diamond.svg'
-import { ReactComponent as Oval } from './assets/oval.svg'
-import { ReactComponent as Tilde } from './assets/tilde.svg'
-import { ReactElement, useState, createElement } from 'react'
+import { CardState } from './App'
+import { InputClickFunction } from './Board'
+import { ReactComponent as Diamond} from './assets/diamond.svg'
+import { ReactComponent as Oval} from './assets/oval.svg'
+import { ReactComponent as Tilde} from './assets/tilde.svg'
+import { FunctionComponent, createElement } from 'react'
 
-const shapeLookup = {
-  'oval': Oval,
-  'diamond': Diamond,
-  'tilde': Tilde
-}
+type CardImage = FunctionComponent<{ color: any; fill: any; }>
+
+const shapeLookup = new Map<string, CardImage>()
+  .set('oval', Oval)
+  .set('diamond', Diamond)
+  .set('tilde', Tilde)
 
-const Card = ({ card, onInputClick }) => {
-  const onInputClickId = (type, value) => onInputClick(card.id, type, value)
+const Card = ({ card, onInputClick }: { card: any, onInputClick: InputClickFunction}) => {
+  const onInputClickId = (type: keyof CardState, value: string | number ) => onInputClick(card.id, type, value)
```

Comment:
this `keyof CardState` bit means that `type` can be any property that exists on the `CardState` class.